src/script/stores/complianceStore.ts

```diff
@@ -0,0 +1,78 @@
+/**
+ * (c) 2023, Center for Computational Thinking and Design at Aarhus University and contributors
+ *
+ * SPDX-License-Identifier: MIT
+ */
+
+import { writable } from 'svelte/store';
+
+// Integrates the Micro:bit Educational Foundation common cookie consent dialog/analytics.
+// Not suitable for other deployments.
+
+export interface CookieConsent {
+  analytics: boolean;
+  functional: boolean;
+}
+
+const domain = window.location.hostname;
+const config = {
+  ga:
+    process.env.VITE_STAGE === 'PRODUCTION' || process.env.VITE_STAGE === 'STAGING'
+      ? {}
+      : undefined,
+  custom: [
+    {
+      type: 'local',
+      domain,
+      category: 'essential',
+      name: 'gestureData',
+      purpose: 'Stores the training data and other settings for each gesture',
+    },
+    {
+      type: 'local',
+      domain,
+      category: 'essential',
+      name: 'btPatternInput',
+      purpose: 'Stores the pairing pattern for the most recent input micro:bit',
+    },
+    {
+      type: 'local',
+      domain,
+      category: 'essential',
+      name: 'btPatternOutput',
+      purpose: 'Stores the pairing pattern for the most recent output micro:bit',
+    },
+    {
+      type: 'local',
+      domain,
+      category: 'essential',
+      name: 'lang',
+      purpose: 'Stores your chosen language',
+    },
+    // Some of the Svelte stores use local storage, this needs investigating
+  ],
+};
+
+function showConsent(
+  { userTriggered }: { userTriggered: boolean } = { userTriggered: false },
+) {
+  const w = window as any;
+  w.commonConsent?.show({ userTriggered, config });
+}
+
+export function manageCookies() {
```

Comment:
This is unused for the moment but there needs to be a way to revoke consent which can be done via the cookie dialog. We should wire this up to a new menu similar to other Foundation apps with other compliance items. We'll leave it unused for now and tackle this in a future PR.